app/src/features/surveys/view/SurveyAttachments.tsx

```diff
@@ -102,6 +97,8 @@ const SurveyAttachments: React.FC<ISurveyAttachmentsProps> = () => {
         onClose={() => {
           setOpenUploadAttachments(false);
           getAttachments(true);
+          console.log('getAttachments');
+          surveyContext.artifactDataLoader.refresh(projectId, surveyId);
```

Comment:
Seems like we have 2 separate calls that fetch attachments here. Looks like the context data loader could replace the `getAttachments` function, and this component could just reference `artifactDataLoader.data`.

This looks like its related to my other comment on `AttachmentsList.tsx`. Probably the changes for both would go hand-in-hand.